rero_ils/modules/items/api/record.py

```diff
@@ -362,15 +377,15 @@ def get_item_by_barcode(cls, barcode, organisation_pid):
         :rtype `rero_ils.modules.items.api.api.Item`
         """
         from . import ItemsSearch
+        filters = Q('term', barcode=barcode)
+        if organisation_pid:
+            filters &= Q('term', organisation__pid=organisation_pid)
         results = ItemsSearch()\
-            .filter('term', barcode=barcode)\
-            .filter('term', organisation__pid=organisation_pid)\
+            .filter('bool', must=[filters])\
```

Comment:
you could write `.filter(filters)` in this case (just discovered and tested yesterday)